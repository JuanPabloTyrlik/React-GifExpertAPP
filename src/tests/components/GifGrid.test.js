import React from 'react';
import '@testing-library/jest-dom';
import { shallow } from 'enzyme';
import GifGrid from '../../components/GifGrid';
import { useFetchGifs } from '../../hooks/useFetchGifs';
jest.mock('../../hooks/useFetchGifs');

describe('Tests on GifGrid', () => {
    const category = 'One Punch Man';
    test('should render correctly', () => {
        useFetchGifs.mockReturnValue({
            data: [],
            loading: true,
        });
        const wrapper = shallow(<GifGrid category={category} />);
        expect(wrapper).toMatchSnapshot();
    });
    test('should render gifs', () => {
        const gifs = [
            {
                id: 'Levi',
                title: 'Levi GIF',
                url: 'https://localhost/gifs/levi.gif',
            },
        ];
        useFetchGifs.mockReturnValue({
            data: gifs,
            loading: false,
        });
        const wrapper = shallow(<GifGrid category={category} />);
        expect(wrapper.find('p').exists()).toBe(false);
        expect(wrapper.find('GifGridItem').length).toBe(gifs.length);
        expect(wrapper).toMatchSnapshot();
    });
});
