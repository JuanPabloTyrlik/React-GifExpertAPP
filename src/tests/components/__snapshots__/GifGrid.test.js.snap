// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests on GifGrid should render correctly 1`] = `
<Fragment>
  <h3>
    One Punch Man
  </h3>
  <p
    className="animate__animated animate__flash"
  >
    Loading...
  </p>
  <div
    className="card-grid"
  />
</Fragment>
`;

exports[`Tests on GifGrid should render gifs 1`] = `
<Fragment>
  <h3>
    One Punch Man
  </h3>
  <div
    className="card-grid"
  >
    <GifGridItem
      id="Levi"
      key="Levi"
      title="Levi GIF"
      url="https://localhost/gifs/levi.gif"
    />
  </div>
</Fragment>
`;
